src/mscorlib/src/System/Random.cs

```diff
@@ -114,11 +121,39 @@ public Random()
           return retVal;
       }
 
+
+      [ThreadStatic]
+      private static Random t_threadRandom;
+      private static Random s_globalRandom = new Random(GenerateGlobalSeed());
+
+      /*=====================================GenerateSeed=====================================
+      **Returns: An integer that can be used as seed values for consecutively
+                 creating lots of instances on the same thread within a short period of time.
+      ========================================================================================*/
+      private static int GenerateSeed() {
+          if (t_threadRandom == null) {
+              int seed;
+              lock (s_globalRandom) {
+                  seed = s_globalRandom.Next();
+              }
+              t_threadRandom = new Random(seed);
+          }
+          return t_threadRandom.Next();
+      }
+
+      /*==================================GenerateGlobalSeed====================================
+      **Action:  Creates a number to use as global seed.
+      **Returns: An integer that is safe to use as seed values for thread-local seed generators.
+      ==========================================================================================*/
+      private static int GenerateGlobalSeed() {
+          return Guid.NewGuid().GetHashCode();
```

Comment:
Process.GetCurrentProcess().PriorityClass = ProcessPriorityClass.High;

            Stopwatch.StartNew().Restart(); //warmup

            var sw = new Stopwatch();

            for (int i = 0; i < 100; i++)
            {
                sw.Restart();
                Guid.NewGuid();
                sw.Stop();
                Console.WriteLine(sw.Elapsed.Ticks);
            }

Gives me:

292
84
75
69
63
57
6
3
3
6
3
3
3
6
3
3
6
3
3
6
3
6
3
6
3
3
6
6
...

x64, Release, no Debugger.Dirty benchmark, but plausible results and repeatable.

So we pay about 2.92 micro seconds for quality random numbers. Environment.TickCount appears to take 9 ticks.